engine/chain/internal/ef/converter_test.go

```diff
@@ -0,0 +1,140 @@
+		},
+		"Convert tx with two inputs from one source": {
+			rawTXBase64: txTwoInputsOneSource,
+			txGetter: newMockTransactionsGetter(t, []string{
+				txTwoInputsOneSourceSource,
+			}),
+			expectedEFHex: "010000000000000000ef02500ba16a6ff2cb7c6a1bb3acaf8558672004cf54dbff86c5f14002549e1f792f010000006a47304402206274b9ad649ec0b3ee6338c9241d8512f0f7c7377f2e0f8e5f104dcc97e5168302200146ce4bb7b07d2b878daf986a7cd13b359d0128feaa375805085a154452ebe041210223466abca89f9a652c3d45435e1b7ce6520a239aefeeb52eae0ae54b08037b31ffffffff00943577000000001976a9140213c4a08b2f2666d16c8c5514358a828420367088ac500ba16a6ff2cb7c6a1bb3acaf8558672004cf54dbff86c5f14002549e1f792f020000006a473044022055f11aa56cc3cdf29cee4a32fe7538ab885d66153d814667e8695f638080413502206dd552b132f036c3194df640bf63f5267bca4bff00392d60a4a135fbef474711412102a1553e5dcb13b6ce5c65bf9e3aee45d09890d89dc9ded2774a20a57b258d294bffffffff00943577000000001976a91474605bc0fcd17b164dcd17b3528c91eaeaee76d388ac107a125183000000001976a91474b772406265b582f72bfa8cf7a434f4163e3c2a88ac00ca9a3b000000001976a9143403fb25f923c2aafe312c1e5e118f8d4a0e290888ac0065cd1d000000001976a9144b4c67d2ec5dd7ea4e6d9f95c68319db9c42cf9488ac00c2eb0b000000001976a914064d8ab782c7438cf647dce4f4107c6f5322c53c88ac80f0fa02000000001976a914d91fe625f566eb8c02b0c0f958faf15aaac633a888ac002d3101000000001976a914b2641346e09cf5dfbd6c91de0fc9eca3425eb34d88ac002d3101000000001976a914bcb8af0f1641192db077c971caae84f8b73adfc588ac404b4c00000000001976a9149ce2d85a530d44270139887b90664e1d2dd241fc88ac40420f00000000001976a914f3d2fc58574a66f1d8ac63f49c3f558d3c35908988ac20a10700000000001976a914fe39ece045b4b22232bc30ae7cc455ccb87aba5988ac400d0300000000001976a914021cf0efb6ec49dd40b3d313e49a5f66a26e895788aca0860100000000001976a914c51da19d540826c982b633ba313ba5b2e7876dc788ac50c30000000000001976a914826bd0a927abe2456988fa362949306c220d924788ac204e0000000000001976a914ff149dfbeec48e8aa2834b9ec174fd83efcf2cbb88ac00000000000000003d006a036478732f757831337033446661313043487068456f6a7064315a513d3d2c6d3136362c74622c61313030302c6537302e3132330474657874014214060000000000001976a914ca0c3030987bafba6d27ce4a0eaf11676d79ddfe88ac00000000",
+		},
+	}
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			tx := txFromBase64(t, test.rawTXBase64)
+
+			converter := ef.NewConverter(test.txGetter)
+			efHex, err := converter.Convert(context.Background(), tx)
+			require.NoError(t, err)
+			require.Equal(t, test.expectedEFHex, efHex)
+
+			// additionally check if converter can convert already-in-ef-hex tx
+			// without the need to fetch source transactions
+			tx, err = sdk.NewTransactionFromHex(efHex)
+			require.NoError(t, err)
+			converter = ef.NewConverter(newMockTransactionsGetter(t, []string{}))
+			efHexRegenerated, err := converter.Convert(context.Background(), tx)
+			require.NoError(t, err)
+			require.Equal(t, efHex, efHexRegenerated)
```

Comment:
for me it should be a single separated test case, to make it clear why it's failing.